### PR TITLE
feat: create organizations slug

### LINKED ISSUE #353: Create organizations handle
In order to improve the searching we should change organizations show page:

Before: /organizations/bdffe466-c316-43f4-83b7-94134d56ea21
After:  /organizations/cesium

What needs to be done:
- [ ] Create a handle for organizations with a unique index
- [ ]  Change routes

You can look into this [PR](https://github.com/cesium/atomic/pull/321) that has a similar solution.

### PR DESCRIPTION
**Pull request by MarioRodrigues10**

### CODE REVIEW

File: lib/atomic/organizations/organization.ex [hunk 1]
```diff
@@ -7,14 +7,15 @@ defmodule Atomic.Organizations.Organization do
   alias Atomic.Organizations.{Announcement, Board, Card, Department, Membership, Partner}
   alias Atomic.Uploaders
 
-  @required_fields ~w(name description)a
+  @required_fields ~w(name handle description)a
```
Comment (feliciofilipe) [inline lib/atomic/organizations/organization.ex]:
```suggestion
  @required_fields ~w(name slug description)a
```
I think `handler` is the term you are looking for. Nevertheless, since `handler` is somewhat of a reserved key word (`handle_event`, `handle_params`, etc) and may cause some confusion, what do you thnk about using `slug` instead? :bulb: 

The correct term on the FE should be something like `username`

Comment (MarioRodrigues10) [reply lib/atomic/organizations/organization.ex]: Done , thanks for the alert !! :+1:

File: lib/atomic/organizations/organization.ex [hunk 2]
```diff
@@ -66,4 +68,30 @@ defmodule Atomic.Organizations.Organization do
...
+    |> case do
+      %{changes: %{slug: _}} = changeset -> changeset
+      %{} = changeset -> add_error(changeset, :slug, "did not change")
+    end
+  end
+
+  defp validate_slug(changeset) do
+    changeset
+    |> validate_required([:slug])
+    |> validate_format(:slug, ~r/^[a-zA-Z0-9_.]+$/,
```
Comment (danielsp45) [inline lib/atomic/organizations/organization.ex:100]: Is it a good idea to allow case sensitive slugs?

Comment (ruilopesm) [reply lib/atomic/organizations/organization.ex:100]: I agree they should be case insensitive. Meaning "cesium" and "CeSIUM" both stand for the same slug.

File: lib/atomic/activities.ex [hunk 3]
```diff
@@ -100,8 +101,11 @@ defmodule Atomic.Activities do
   """
   def get_session_organizations!(session, preloads \\ [:departments]) do
```
Comment (ruilopesm) [inline lib/atomic/activities.ex]: So this function is called `get_session_organizations!`, but it returns slugs? 🤔

File: lib/atomic/organizations.ex [hunk 4]
```diff
@@ -81,6 +81,31 @@ defmodule Atomic.Organizations do
     |> Repo.update()
   end
 
+  @doc """
+  Gets an organization by slug.
+
+  ## Examples
+      iex> get_organization_by_slug("foo_bar")
+      %Organization{}
```
Comment (ruilopesm) [inline lib/atomic/organizations.ex:129]:
```suggestion
      %Organization{}
      
```

File: lib/atomic/organizations.ex [hunk 5]
```diff
@@ -213,7 +238,7 @@ defmodule Atomic.Organizations do
       nil
 
   """
-  def get_role(user_id, organization_id) do
+  def get_role(user_id, organization_id) when is_binary(user_id) and is_binary(organization_id) do
```
Comment (ruilopesm) [inline lib/atomic/organizations.ex:289]: Why this change?

File: lib/atomic_web/live/activity_live/index.html.heex [hunk 6]
```diff
@@ -10,7 +10,7 @@
         <%= if not @empty and @has_permissions do %>
           <div class="hidden lg:border-orange-500 lg:block">
             <%= live_patch("+ New Activity",
-              to: Routes.activity_new_path(@socket, :new, @current_organization),
+              to: Routes.activity_new_path(@socket, :new, @current_organization.slug),
```
Comment (ruilopesm) [inline lib/atomic_web/live/activity_live/index.html.heex:14]: > Additionally, interpolated ~p values are encoded via the [Phoenix.Param](https://hexdocs.pm/phoenix/Phoenix.Param.html) protocol. For example, a %Post{} struct in your application may derive the [Phoenix.Param](https://hexdocs.pm/phoenix/Phoenix.Param.html) protocol to generate slug-based paths rather than ID based ones. This allows you to use ~p"/posts/#{post}" rather than ~p"/posts/#{post.slug}" throughout your application. See the [Phoenix.Param](https://hexdocs.pm/phoenix/Phoenix.Param.html) documentation for more details.

What about implementing this? Then we would only need to write `@current_organization` instead of `@current_organization.slug` 💡

File: lib/atomic_web/live/activity_live/show.ex [hunk 7]
```diff
@@ -7,35 +7,37 @@ defmodule AtomicWeb.ActivityLive.Show do
...
     if connected?(socket) do
       Activities.subscribe("new_enrollment")
       Activities.subscribe("deleted_enrollment")
     end
 
-    {:ok, assign(socket, :id, id)}
+    {:ok,
+     socket
+     |> assign(:id, id)
+     |> assign(:slug, slug)}
```
Comment (ruilopesm) [inline lib/atomic_web/live/activity_live/show.ex:18]: Why are these assigns needed?

File: lib/atomic_web/live/membership_live/index.html.heex [hunk 8]
```diff
@@ -41,7 +41,7 @@
                   <td class="whitespace-nowrap px-3 py-4 text-sm text-gray-500"><%= membership.created_by.name %></td>
                   <td class="whitespace-nowrap px-3 py-4 text-sm text-gray-500"><%= display_date(membership.updated_at) %> <%= display_time(membership.updated_at) %></td>
                   <td class="relative whitespace-nowrap py-4 pl-3 pr-4 text-right text-sm font-medium sm:pr-6">
-                    <%= link to: Routes.membership_edit_path(@socket, :edit, membership.organization_id, membership), class: "text-indigo-600 hover:text-indigo-900" do %>
+                    <%= link to: Routes.membership_edit_path(@socket, :edit, @current_organization.handle, membership), class: "text-indigo-600 hover:text-indigo-900" do %>
```
Comment (ruilopesm) [inline lib/atomic_web/live/membership_live/index.html.heex]:
```suggestion
                    <%= link to: Routes.membership_edit_path(@socket, :edit, @current_organization.slug, membership), class: "text-indigo-600 hover:text-indigo-900" do %>
```

File: priv/repo/migrations/2022000000000_create_organizations.exs [hunk 9]
```diff
@@ -13,5 +14,6 @@ defmodule Atomic.Repo.Migrations.CreateOrganizations do
     end
 
     create unique_index(:organizations, [:name])
```
Comment (ruilopesm) [inline priv/repo/migrations/2022000000000_create_organizations.exs:16]: We don't need an index by name. I know it was not added now, but, please, remove it 🙏🏼

File: lib/atomic/activities.ex [hunk 10]
```diff
@@ -141,7 +142,27 @@ defmodule Atomic.Activities do
...
+        ** (Ecto.NoResultsError)
+  """
+  def get_activity_organizations!(activity, preloads \\ [:departments]) do
+    Repo.preload(activity, preloads)
+    |> Map.get(:departments, [])
+    |> Enum.map(& &1.organization_id)
+    |> Enum.map(fn id ->
+      Organizations.get_organization!(id).slug
+    end)
+  end
```
Comment (ruilopesm) [inline lib/atomic/activities.ex:162]: We don't need this anymore, since an activity is, for now, associated to only one organization.

File: lib/atomic_web/config.ex [hunk 11]
```diff
@@ -96,6 +99,33 @@ defmodule AtomicWeb.Config do
...
+        icon: :user_add,
+        url: Routes.membership_index_path(conn, :index, current_organization.slug),
+        tabs: []
+      },
+      %{
+        key: :board,
+        title: "Board",
+        icon: :users,
+        url: Routes.board_index_path(conn, :index, current_organization),
+      },
```
Comment (ruilopesm) [inline lib/atomic_web/config.ex:128]: These are admin pages. Why are you adding them here?

File: lib/atomic_web/live/activity_live/index.html.heex (hunk 6 above)
Comment (ruilopesm) [reply lib/atomic_web/live/activity_live/index.html.heex:14]: Don't forget to implement this 🙏🏼

**Comment by diogogmatos** [comment]
[Create Organizations Slug](https://trello.com/c/awPXHgXl/3-create-organizations-slug)

**Comment by diogogmatos** [comment]
[Create Organizations Slug](https://trello.com/c/awPXHgXl/3-create-organizations-slug)

**Comment by joaodiaslobo** [comment]
Will be done in the future
